src/develop/blend.c

```diff
@@ -220,6 +223,65 @@ int dt_develop_blendif_init_masking_profile(struct dt_dev_pixelpipe_iop_t *piece
   return 1;
 }
 
+static float _detail_mask_threshold(const float level, const gboolean detail)
+{
+  // this does some range calculation for smoother ui experience
+  return 0.01f * (detail ? powf(level, 1.5f) : 1.0f - powf(fabs(level), 0.75f ));
+}
+
+static void _refine_with_detail_mask(struct dt_iop_module_t *self, struct dt_dev_pixelpipe_iop_t *piece, float *mask, const struct dt_iop_roi_t *const roi_in, const struct dt_iop_roi_t *const roi_out, const float level)
+{
+  if(level == 0.0f) return;
+
+  const gboolean detail = (level > 0.0f);
+  const float threshold = _detail_mask_threshold(level, detail);
+  
+  float *tmp = NULL;
+  float *lum = NULL;
+  float *warp_mask = NULL;
+
+  dt_dev_pixelpipe_t *p = piece->pipe;
+  if(p->luminance_mask_data == NULL) return;
+
+  const int iwidth  = p->luminance_mask_roi.width;
+  const int iheight = p->luminance_mask_roi.height;
+  const int owidth  = roi_in->width;
+  const int oheight = roi_in->height;
+
+  const int bufsize = MAX(iwidth * iheight, owidth * oheight);
+  
+  tmp = dt_alloc_align_float(bufsize);
+  lum = dt_alloc_align_float(bufsize);
+  if((tmp == NULL) || (lum == NULL)) goto error;
+
+  dt_masks_calc_detail_mask(p->luminance_mask_data, lum, tmp, iwidth, iheight, threshold, detail);
+
+  // here we have the slightly blurred full detail mask available
+  warp_mask = dt_dev_distort_luminance_mask(p, lum, self);
+  if(warp_mask == NULL) goto error;
```

Comment:
we can free memory for lum & tmp just above if. And the remove them below.

Reply:
ok, lowers mem pressure